### PR TITLE
fix(server.display-name): display loader during state navigation

### PR DESCRIPTION
**Pull request by anooparveti**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-8749
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~ [n/a]
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~ [n/a]

## Description

 Display loader during state navigation after server name update

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicated/server/display-name/display-name.controller.js [hunk 1]
```diff
@@ -61,8 +56,16 @@ angular.module('App').controller(
...
+            '^',
+            {},
+            {
+              reload,
+            },
+          ),
+        )
         .finally(() => {
           this.loading.update = false;
         });
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/server/display-name/display-name.controller.js:71]: Maybe we can merge the 2 .finally()?

**Review by ghyenne — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/dedicated/server/display-name/display-name.controller.js [hunk 2]
```diff
@@ -61,8 +56,16 @@ angular.module('App').controller(
...
         })
+        .finally(() =>
+          this.$state.go(
+            '^',
+            {},
+            {
+              reload,
+            },
+          ),
+        )
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicated/server/display-name/display-name.controller.js:68]: @anooparveti, for migration raisons, we have to not implement $state in controllers.
If duable you have to write it into a routing file.

Also the reload have to be implement into then promise.
```suggestion
        .then(() =>
          this.refreshPage(),
        ).catch(() => {  })
```

File: packages/manager/apps/dedicated/client/app/dedicated/server/display-name/display-name.controller.js (hunk 1 above)
Comment (anooparveti) [reply packages/manager/apps/dedicated/client/app/dedicated/server/display-name/display-name.controller.js:71]: The second `finally` depends on the promise returned by `$state.go`. So they both can not be combined.

File: packages/manager/apps/dedicated/client/app/dedicated/server/display-name/display-name.controller.js (hunk 2 above)
Comment (anooparveti) [reply packages/manager/apps/dedicated/client/app/dedicated/server/display-name/display-name.controller.js:68]: Hello @mohammed-zahaf, I agree with your suggestion of moving `$state` to routing file but unfortunately this particular state is legacy and is implemented without `component` architecture. So it cannot be implemented as of now.

**Review by radireddy — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

**Pull request merged**